### PR TITLE
Mob test build 

### PR DESCRIPTION
**Pull request by stevenleej**
Added enemy model and animations

Added enemy 1's behaviour tree and animator

switched player movement to switch statements for certain conditions to ease readability

### CODE REVIEW

File: The_Dune_Project/Assets/Scripts/Fight/EnemyController.cs [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
+    [SerializeField] protected Vector3 homeLocation;
+    [SerializeField] protected NavMeshAgent myAgent;
+    [SerializeField] protected Fighter myFighter;
     
     [ReadOnlyAttribute, SerializeField] protected Transform target;
-
-    private void Start()
+    
+    
+    protected void LoadAgentProperties(EnemyController e)
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Fight/EnemyController.cs:25]: you dont need the parameter EnemyController e

File: The_Dune_Project/Assets/Scripts/Fight/MeleeEnemyController.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    
+    private void Start()
+    {
+        LoadAgentProperties(this);
+        enemyAnimatorManager = GetComponent<EnemyAnimatorManager>();
+    }
+
+    private void Update()
+    {
+        base.Update();
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Fight/MeleeEnemyController.cs:21]: unnecessary func

File: The_Dune_Project/Assets/Scripts/Fight/MeleeEnemyController.cs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        if (targetFighter != null && hitCooldownTimer <= 0.0f)
+        {
+            enemyAnimatorManager.SetAnimBool("walking", false);
+            enemyAnimatorManager.PlayTargetAnimation("slash", true);
+            //perform knockback to the player as an aniation event
+            hitCooldownTimer = hitCooldown;
+        }
+    }
+
+    public void ReturnToOriginalLocation()
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Fight/MeleeEnemyController.cs:79]: when is this called?

File: The_Dune_Project/Assets/Scripts/Fight/MeleeEnemyController.cs [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            isAggroed = false;
+            myAgent.SetDestination(homeLocation);
+            enemyAnimatorManager.SetAnimBool("walking", false);
+        }
+    }
+
+    public bool PlayerIsInRangeOfMe()
+    {
+        float distance = Vector3.Distance(target.position, transform.position);
+        if (distance <= lookRadius)
```
Comment (Vassenx) [inline The_Dune_Project/Assets/Scripts/Fight/MeleeEnemyController.cs:92]: can just do
return distance <= lookRadius
although its less readable
nitpick ^

**Review by Vassenx — CHANGES_REQUESTED** [review]

File: The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs [hunk 5]
```diff
@@ -102,11 +105,11 @@ public void HandleAllPlayerMovement()
...
         HandleGravity();  
         HandleFalling(); 
-        HandleCCJumping();
-        if(!playerHookHandler.finishedHook){
+        if (playerManager.isInteracting){
             return;
         }
-        if (playerManager.isInteracting){
+        HandleCCJumping();
+        if(!playerHookHandler.finishedHook){
```
[the remark this answers is not in the document]
Comment (stevenleej) [reply The_Dune_Project/Assets/Scripts/Player/PlayerMovement.cs:112]: interacting skips the player movement loop; for example, when a player's animation parameter is set to isInteracting, relevant code which comes after does not run (in our case, moving and turning)

File: The_Dune_Project/Assets/Scripts/Fight/MeleeEnemyController.cs (hunk 2 above)
Comment (stevenleej) [reply The_Dune_Project/Assets/Scripts/Fight/MeleeEnemyController.cs:21]: you need to call it or unity won't update the cooldowns on the mob (monobehaviour and inheritance with update functions is stupid)

File: The_Dune_Project/Assets/Scripts/Fight/MeleeEnemyController.cs (hunk 3 above)
Comment (stevenleej) [reply The_Dune_Project/Assets/Scripts/Fight/MeleeEnemyController.cs:79]: to go back to the original location in the behaviour tree (assuming it's the return to original location function)

**Pull request merged**